File: CSHTML5.Samples.Showcase/Other/MaterialDesign_Styles_Kit/AccentColorConverter.cs
using System;
using Windows.UI;
using Windows.UI.Xaml.Data;
using Windows.UI.Xaml.Media;

namespace MaterialDesign_Styles_Kit
{
    public class AccentColorConverter : IValueConverter
    {
        public object Convert(object value, Type targetType, object parameter, string language)
        {
            if (value is SolidColorBrush)
            {
                Color oldColor = ((SolidColorBrush)value).Color;
                Byte newR = oldColor.R > 40 ? (Byte)(oldColor.R - 40) : (Byte)0;
                Byte newG = oldColor.G > 40 ? (Byte)(oldColor.G - 40) : (Byte)0;
                Byte newB = oldColor.B > 40 ? (Byte)(oldColor.B - 40) : (Byte)0;
                Color newColor = Color.FromArgb(oldColor.A, newR, newG, newB);
                return new SolidColorBrush(newColor);
            }
            return value;
        }

        public object ConvertBack(object value, Type targetType, object parameter, string language)
        {
            throw new NotImplementedException();
        }
    }
}

File: vbsrc/Other/MaterialDesign_Styles_Kit/AccentColorConverter.vb
Imports System

#If SLMIGRATION
Imports System.Windows.Data
Imports System.Windows.Media
Imports System.Globalization
#Else
using Windows.UI;
using Windows.UI.Xaml.Data;
using Windows.UI.Xaml.Media;
#End If

Namespace Global.MaterialDesign_Styles_Kit
    Public Class AccentColorConverter
        Implements IValueConverter
#If SLMIGRATION Then
#Else
        public object Convert(object value, Type targetType, object parameter, string language)
#End If
        Public Function Convert(ByVal value As Object, ByVal targetType As Type, ByVal parameter As Object, ByVal culture As CultureInfo) As Object Implements IValueConverter.Convert
            If TypeOf value Is SolidColorBrush Then
                Dim oldColor = CType(value, SolidColorBrush).Color
                Dim newR = If(oldColor.R > 40, CByte(oldColor.R - 40), CByte(0))
                Dim newG = If(oldColor.G > 40, CByte(oldColor.G - 40), CByte(0))
                Dim newB = If(oldColor.B > 40, CByte(oldColor.B - 40), CByte(0))
                Dim newColor = Color.FromArgb(oldColor.A, newR, newG, newB)
                Return New SolidColorBrush(newColor)
            End If
            Return value
        End Function

#If SLMIGRATION Then
#Else
        public object ConvertBack(object value, Type targetType, object parameter, string language)
#End If
        Public Function ConvertBack(ByVal value As Object, ByVal targetType As Type, ByVal parameter As Object, ByVal culture As CultureInfo) As Object Implements IValueConverter.ConvertBack
            Throw New NotImplementedException()
        End Function
    End Class
End Namespace
